R/ggplot2-scales.R

```diff
@@ -115,3 +115,112 @@ To avoid this error, use a larger palette or `interpolate = TRUE`.",
     ggplot2::scale_fill_gradientn(colours = pal(256), ...)
   }
 }
+
+
+#' Create theme for nmfs colors
+#' @inheritParams scale_color_nmfs
+#' @param base_size The base font size, as defined in the
+#'   \code{\link[ggplot2]{theme_gray}} function
+#' @param lab_size The axis text size
+#' @param ink The plot foreground color, as defined in the
+#'   \code{\link[ggplot2]{theme_gray}} function
+#' @param paper The plot background color, as defined in the
+#'   \code{\link[ggplot2]{theme_gray}} function
+#' @param accent The plot's accented elements' color, as defined in the
+#'   \code{\link[ggplot2]{theme_gray}} function
+#' @param ... Additional arguments passed to: [ggplot2::scale_fill_gradientn()]
+#' when `discrete` is TRUE; [ggplot2::discrete_scale()] when `discrete` is FALSE
+#' and `interpolate` is TRUE; and [ggplot2::scale_fill_manual()] when `discrete`
+#' is FALSE and `interpolate` is FALSE.
+#' @return A theme that applies nmfs color palettes and theme-related elements (such as label sizes, borders, axis lines, and more) to a plot. For more information about the ggplot2 advances that enabled this function, check out the [ggplot2 version 4.0.0 release notes](https://tidyverse.org/blog/2025/09/ggplot2-4-0-0/).
+#' @examples
+#' library(ggplot2)
+#' ggplot(mtcars, aes(x = mpg, y = disp, color = as.factor(cyl))) +
+#'   geom_point(size = 3) +
+#'   theme_nmfs(discrete = TRUE, interpolate = TRUE, palette = "crustacean")
+#' ggplot(mtcars, aes(x = mpg, y = disp, color = as.factor(cyl))) +
+#'   geom_point(size = 3) +
+#'   theme_nmfs(discrete = TRUE, interpolate = FALSE, palette = "crustacean")
+#' ggplot(mtcars, aes(x = mpg, y = disp, fill = hp)) +
+#' geom_point(size = 3, shape = 24) +
+#' theme_nmfs(discrete = FALSE, interpolate = TRUE, palette = "seagrass")
+#' @export
+theme_nmfs <- function(
+    palette = "oceans", 
+    discrete = TRUE,
+    reverse = FALSE,
+    interpolate = TRUE,
+    base_size = 14,
+    lab_size = 12,
+    ink = "black",
+    paper = "white",
+    accent = "#003087",
+    ...) {
+  rlang::warn(
```

Comment:
I think this warning happens once for everyone (please correct me if I am wrong here!) - I wonder if instead it would be more useful to error with a helpful message for people using a lower version of ggplot2 only? 

Also, a pet peeve of mine - utilize -> use. They have identical meanings, but utilize is more complicated. Please ignore if you have different opinions :)